src/TTextCodec.cpp

```diff
@@ -0,0 +1,738 @@
+    QChar(0x2514), QChar(0x2534), QChar(0x252C), QChar(0x251C), QChar(0x2500), QChar(0x253C), QChar(0x03A0), QChar(0x03A1),  // C0-C7
+    QChar(0x255A), QChar(0x2554), QChar(0x2569), QChar(0x2566), QChar(0x2560), QChar(0x2550), QChar(0x256C), QChar(0x03A3),  // C8-CF
+    QChar(0x03A4), QChar(0x03A5), QChar(0x03A6), QChar(0x03A7), QChar(0x03A8), QChar(0x03A9), QChar(0x03B1), QChar(0x03B2),  // D0-D7
+    QChar(0x03B3), QChar(0x2518), QChar(0x250C), QChar(0x2588), QChar(0x2584), QChar(0x03B4), QChar(0x03B5), QChar(0x2580),  // D8-DF
+    QChar(0x03B6), QChar(0x03B7), QChar(0x03B8), QChar(0x03B9), QChar(0x03BA), QChar(0x03BB), QChar(0x03BC), QChar(0x03BD),  // E0-E7
+    QChar(0x03BE), QChar(0x03BF), QChar(0x03C0), QChar(0x03C1), QChar(0x03C3), QChar(0x03C2), QChar(0x03C4), QChar(0x0384),  // E8-EF
+    QChar(0x00AD), QChar(0x00B1), QChar(0x03C5), QChar(0x03C6), QChar(0x03C7), QChar(0x00A7), QChar(0x03C8), QChar(0x0385),  // F0=F7
+    QChar(0x00B0), QChar(0x00A8), QChar(0x03C9), QChar(0x03CB), QChar(0x03B0), QChar(0x03CE), QChar(0x25A0), QChar(0x00A0)}; // F8-FF
+// clang-format on
+
+// We give ours a distict different name with an M_ prefix so we can tell it
+// apart from a system one
+QByteArray TTextCodec_437::name() const
+{
+    return "M_CP437";
+}
+
+QByteArray TTextCodec_667::name() const
+{
+    return "M_CP667";
+}
+
+QByteArray TTextCodec_737::name() const
+{
+    return "M_CP737";
+}
+
+QByteArray TTextCodec_869::name() const
+{
+    return "M_CP869";
+}
+
+// Data for the following two types of methods taken from entry in:
+// http://www.iana.org/assignments/character-sets/character-sets.xml
+QList<QByteArray> TTextCodec_437::aliases() const
+{
+    return {"IBM437", "437", "cp437", "csPC8CodePage437"};
+}
+
+int TTextCodec_437::mibEnum() const
+{
+    return 2011; // Same as IBM437
+}
+
+QList<QByteArray> TTextCodec_667::aliases() const
+{
+    return {"Mazovia", "667", "cp6677"};
+}
+
+int TTextCodec_667::mibEnum() const
+{
+    return 2999; // Vendor, and used just to give a number
+}
+
+QList<QByteArray> TTextCodec_737::aliases() const
+{
+    return {"737", "cp737"};
+}
+
+int TTextCodec_737::mibEnum() const
+{
+    return 2998; // Vendor, and used just to give a number
+}
+
+QList<QByteArray> TTextCodec_869::aliases() const
+{
+    return {"IBM869", "ibm-869", "869", "cp869", "csIBM869", "ibm-869_P100-1995", "windows-869", "cp-gr"};
+}
+
+int TTextCodec_869::mibEnum() const
+{
+    return 2054;
+}
+
+// All 8-bit values are valid for conversion from CP437 to Unicode so there will
+// never be any invalid characters or any left over that cannot be converted and
+// need to wait for the next chunk of data in the next call. However IF we have
+// a non-null ConverterState pointer we ought to consider the first of the three
+// int state_data[3] array and if castable to a (bool) false, i.e. zero we
+// should insert a BOM if the IgnoreHeader flag is not set. If there is no state
+// we should insert the BOM for each call:
```

Comment:
I don't know too much about this - but isn't using BOM a no-no?

Reply:
Depends on what the caller asks for, if there is a pointed to `ConverterState` and that has the flag `QTextCodec::IgnoreHeader` set in it's `QTextCodec::ConverterState::flags` member then yes we can ignore/not produce a BOM but not otherwise...

Reply:
This is one of the things that was missing from the samples of other peoples attempts at producing `QTextCodec` derived class and if someone else comes along and reuses *my* code I'd like to think they'd have a more complete example to work from. :innocent: